odata-protocol/11.4 Data Modification.md

```diff
@@ -1078,18 +1088,31 @@ If an individual change fails due to a failed dependency, it MUST be
 annotated with the term [`Core.DataModificationException`](https://github.com/oasis-tcs/odata-vocabularies/blob/main/vocabularies/Org.OData.Core.V1.md#DataModificationException) and SHOULD specify
 a `responseCode` of `424` ([Failed Dependency](#ResponseCode424FailedDependency)).
 
-Alternatively, the verb `PUT` can be used, in which case the request
-body MUST be the representation of a collection of entities. In this
+### ##subsubsec Replace a Collection of Entities
+
+Collections of entities can be replaced by submitting a `PUT` request
+to the resource path of the collection. The body of the request MUST be
+the representation of the complete collection of replacement entities. In this
 case all entities provided in the request are applied as
 [upserts](#UpsertanEntity), and any entities not provided in the request
-are deleted. In this case, if the `continue-on-error` preference has
-been specified, and the request returns a success response code, then a
-response MUST be returned regardless of the
+are deleted. 
+
+For each entity being updated, clients MAY specify an [ETag](#UseofETagsforAvoidingUpdateConflicts)
+value obtained from a previous request. If an ETag is provided that does not match the ETag 
+value of the entity being updated, or if an ETag is provided for an entity that does not 
+currently exist, then services that support ETags MUST NOT apply the 
+change and instead report a `412 Precondition Failed` error according to the appropriate [`continue-on-error`](#ErrorHandlingwhenReplacingaCollectionofEntities) behavior. 
+The special ETag value `*` can be used to match any existing entity but fail if the entity does not already exist.
+
+#### ##subsubsubsec Error Handling when Replacing a Collection of Entities
+
+If the `continue-on-error` preference has been applied and any errors occur 
+in processing the changes, then a response MUST be returned regardless of the
 [`return`](#Preferencereturnrepresentationandreturnminimal) preference, and MUST
 contain the full membership and values of the collection as it exists in
 the service.
```

Comment:
How are skipped data modifications expressed in this full membership response?

We either have to add adapted copies of the relevant parts of the sibling subsubsection for "... Updating ..." or extract the common parts into separate subsub(sub?)sections.

Reply:
By showing the current values -- the section on returning a "compensating" delta payload with data modification exceptions was specific to the delta update case.

Agree that this could be fleshed out more, but I think current wording is as complete as previous wording, and less confusing (because the reader doesn't have to guess which parts apply to PATCH versus PUT).